src/ai/backend/manager/services/agent/service.py

```diff
@@ -137,53 +145,84 @@ async def sync_agent_registry(
 
         return SyncAgentRegistryActionResult(result=None, agent_data=agent_data)
 
-    async def get_watcher_status(
-        self, action: GetWatcherStatusAction
-    ) -> GetWatcherStatusActionResult:
-        watcher_info = await self._get_watcher_info(action.agent_id)
+    async def _request_watcher(
+        self,
+        agent_id: AgentId,
+        method: Literal["GET", "POST"],
+        endpoint: str,
+    ) -> dict:
+        watcher_info = await self._get_watcher_info(agent_id)
         connector = aiohttp.TCPConnector()
+
         async with aiohttp.ClientSession(connector=connector) as sess:
-            # TODO: Ugly naming?
             with _timeout(5.0):
+                watcher_url = watcher_info["addr"] / endpoint
                 headers = {"X-BackendAI-Watcher-Token": watcher_info["token"]}
-                async with sess.get(watcher_info["addr"], headers=headers) as resp:
-                    return GetWatcherStatusActionResult(resp, agent_id=action.agent_id)
+
+                match method:
+                    case "GET":
+                        request_method = sess.get
+                    case "POST":
+                        request_method = sess.post
```

Comment:
Please create a way to specify the method through sess.request rather than choosing between sess.get and post.